### ISSUE TITLE
String size validator bug

### ISSUE THREAD

**Issue by rlopzc** [ozziexsh/validate#24]
```elixir
iex(1)> Validate.validate("ABCD010101ABCDEM1S", size: 18)
{:error,
 [
   %Validate.Validator.Error{
     path: [],
     message: "must have a size of exactly 18",
     rule: :size
   }
 ]}
iex(2)> String.length("ABCD010101ABCDEM1S")
18
```

**Comment by rlopzc** [comment]
The error is here https://github.com/ozziexsh/validate/blob/main/lib/validate/rules/size.ex#L10-L13 you are only checking if the result is equal to the input (`18 == "ABCD010101ABCDEM1S"`)

**Comment by ozziexsh** [comment]
Thanks! That should've been compared to `x` not `value`. I've fixed this in #25 and published v1.3.1
